### ISSUE TITLE
No option for Punjabi Shahmukhi in Transifex

### ISSUE THREAD

**Issue by bgo-eiu** [openstreetmap/iD#9243]
I tried requesting this in there a few months ago but this option still looks to be missing. Preferably the language code should be `pnb`, but if that is not available because of CLDR or something like that, `pa_PK` would be the next best option since that is what StreetComplete has to use for this (Android doesn't let you pick anything else).

I'm hesitant to add anything to the Gurmukhi translation strings without this as it would be more efficient to add translations in the two scripts at the same time, and be useful to more people since it's a minority of Punjabi speakers that read Gurmukhi.

**Comment by bgo-eiu** [comment]
It may be possible to insert both strings in the single Punjabi field in Transifex, but this is not ideal as one script is right-to-left and the other is left-to-right. It may depend on the browser or other factors, but sometimes I see things like text getting printed backwards because of scripts with different directions being used on the same line

**Comment by 1ec5** [comment]
> I tried requesting this in there a few months ago but this option still looks to be missing.

You aren’t alone in waiting: the iD project in Transifex currently has a backlog of 32 pending requests to start translating into new languages[^requests] and 70 unresolved issues. The iD Editor team there has six users with the Team Manager role, but I’m not sure who among them is still active. As a translator, I can only view these statistics, but maybe @tyrasd can help here.

> Preferably the language code should be `pnb`, but if that is not available because of CLDR or something like that, `pa_PK` would be the next best option since that is what StreetComplete has to use for this (Android doesn't let you pick anything else).

Transifex maintains its own locale list similar to CLDR’s but not quite the same. A Transifex project maintainer would be able to add the following languages:

* Panjabi (Punjabi) (`pa`)
* Panjabi (Punjabi) (India) (`pa_IN`) – hidden by default

Unfortunately, there’s no `pnb` locale as in CLDR. That said, as far as Transifex is concerned, you could put Gurmukhi Punjabi into the Klingon localization and they wouldn’t care, as long as the plural and gender options line up. iD’s Transifex script would just map one locale code to another when pulling translations, as in #4783. One option could be to put Gurmukhi in `pa_IN` and Shahmukhi in `pa`, but it would be important to communicate the convention to current and prospective translators, setting aside the literal meaning of these locale codes.

iD is sort of at the mercy of the operating system when it comes to which locale code it ultimately maps to a given localization. [By default](https://en.wikipedia.org/wiki/Content_negotiation), iD tries to present a localization that matches the browser’s language settings. Browser vendors maintain their own locale lists, but the default locale is influenced by the operating system’s language settings and the browser’s own UI localization. The major browsers and operating systems have historically differed about many different locale codes, but they’ve been converging thanks to CLDR. Firefox offers `pa`, `pa-IN`, and `pa-PK` locales; its Gurmukhi localization uses `pa-IN`, while `pa-PK` would be for Shahmukhi.

[^requests]: This figure likely overstates the backlog. Many of the requests may be by users who want iD in a language but aren’t necessarily volunteering to translate. Many are for country-qualified locales, which normally get declined unless the unqualified locale is already fully translated and there’s an actual need to essentially fork the localization.

**Comment by tyrasd** [comment]
Thanks for the ping, and sorry for not noticing the request.

I have now added @bgo-eiu as a translator for `pa_PK`, but the language still doesn't show up on the transifex user interface, as far as I can see. Maybe you first have to accept the invitation for it to show up?!

<hr>

>  32 pending requests […] and 70 unresolved issues

:unamused:  Unfortunately, I don't have sufficient access rights to clear any unneeded language requests such as `de_DE` for example which would just be a duplicate of the existing `DE` locale. I also cannot resolve the raised issues, most of which are just regular comments after a first glimpse into a sample of them.

@simonpoole are you still [managing](https://lists.openstreetmap.org/pipermail/dev/2014-September/028058.html) the openstreetmap organization on transifex? If yes, could you allow me to properly manage the `id-editor` project there?

**Comment by simonpoole** [comment]
@tyrasd what's your transifex login?

**Comment by tyrasd** [comment]
It's [`tyr_asd`](https://www.transifex.com/user/profile/tyr_asd/)

**Comment by bhousel** [comment]
I've just added tyr_asd and 1ec5 to the list of maintainers for id-editor on Transifex.

Traditionally we have declined requests for translation with a language-culture if we already have a translation for the language.  (e.g. we already have a `de` so we would decline a request for `de-DE`).  I've found that in most cases, people make these requests by accident.

In a few cases it makes sense to have separate culture translations, like `en`, `en-GB`, `en-AU`.
